### ISSUE TITLE
Title: Ruff 0.11.10 Fails to Detect F-string SyntaxError with target-version = "py39"

### ISSUE THREAD

**Issue by tmarquart** [astral-sh/ruff#18194]
### Summary

Ruff Version: 0.11.10
Python Runtime Causing Error: Python 3.11.x (Error also occurs on 3.9.x)
Operating System: Windows
Bug Description:
Ruff does not flag an f-string that reuses the same outer quote character for an inner string literal (e.g., f'...{dict['key']}...'), which is a SyntaxError before Python 3.12. This occurs even when target-version = "py39" is set and standard syntax error rules (like E902 ParseError) are enabled.

To Reproduce:

TOML:
```toml
target-version = "py39"
```

# test_fstring.py
```py
base_type = "sometype"
class DummyType:
    def __init__(self, length=None):
        if length is not None:
            self.length = length
col_props = {'type': DummyType(length=10)}
# Problematic line:
final_type = f'{base_type}({col_props['type'].length})' if hasattr(col_props['type'], 'length') else base_type
# print(f"Final type: {final_type}") # Optional: for running the file
```

```
Runtime Error:
  File "<string>", line 14
    final_type = f'{base_type}({col_props['type'].length})' if hasattr(col_props['type'], 'length') else base_type
                                           ^^^^
SyntaxError: f-string: unmatched '['
```

Behavior confirmed on playground:

Everything is looking good!

Expected Ruff Behavior:
Ruff should report a syntax error (e.g., E902 ParseError or E999 SyntaxError) for the problematic f-string line, consistent with target-version = "py39".

### Version

0.11.10

**Comment by tmarquart** [comment]
Perfect, that is exactly what I needed, I added preview=true and it caught it. I truly appreciate the lightning quick response, thanks so much
